coev/evidence_id.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2025 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package coev
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/veraison/corim/comid"
+	"github.com/veraison/corim/encoding"
+	"github.com/veraison/corim/extensions"
+)
+
+// EvidenceID stores an evidence identity. The supported formats as of now are UUID.
+type EvidenceID struct {
+	Value IEvidenceValue
+}
+
+// NewEvidenceID creates a new evidence with the value of the specified type
+// populated using the provided value.
+func NewEvidenceID(val any, typ string) (*EvidenceID, error) {
+	factory, ok := evidenceIDValueRegister[typ]
+	if !ok {
+		return nil, fmt.Errorf("unknown EvidenceID type: %s", typ)
+	}
+
+	return factory(val)
+}
+
+// Valid checks for the validity of given EvidenceID
+func (o EvidenceID) Valid() error {
+	if o.String() == "" {
+		return errors.New("no EvidenceID")
+	}
+	_, err := o.GetUUID()
+	if err != nil {
+		return fmt.Errorf("unable to fetch valid UUID: %w", err)
+	}
+	return nil
+}
+
+// String returns a printable string of the EvidenceID value.  UUIDs use the
+// canonical 8-4-4-4-12 format.
+func (o EvidenceID) String() string {
+	if o.Value == nil {
+		return ""
+	}
+
+	return o.Value.String()
+}
+
+// Type returns a string naming the type of the underlying EvidenceID value.
+func (o EvidenceID) Type() string {
+	return o.Value.Type()
+}
+
+// Bytes returns a []byte containing the bytes of the underlying EvidenceID
+// value.
+func (o EvidenceID) Bytes() []byte {
+	return o.Value.Bytes()
+}
+
+// MarshalCBOR serializes the target EvidenceID to CBOR
+func (o EvidenceID) MarshalCBOR() ([]byte, error) {
+	return em.Marshal(o.Value)
+}
+
+func (o *EvidenceID) UnmarshalCBOR(data []byte) error {
+	return dm.Unmarshal(data, &o.Value)
+}
+
+// UnmarshalJSON deserializes the supplied JSON object into the target EvidenceID
+// The evidence object must have the following shape:
+//
+//	{
+//	  "type": "<EVIDENCE_TYPE>",
+//	  "value": <EVIDENCE_VALUE>
+//	}
+//
+// where <EVIDENCE_TYPE> must be one of the known IEvidenceValue implementation
+// type names (available in the base implementation: "uuid"), and
+// <EVIDENCE_VALUE> is the JSON encoding of the evidence value. The exact
+// encoding is <EVIDENCE_TYPE> dependent. For the base implmentation types it is
+//
+//	uuid: standard UUID string representation, e.g. "550e8400-e29b-41d4-a716-446655440000"
+
+//nolint:dupl
```

Comment:
curiosity: why is this needed?

Reply:
Is required as otherwise, the `linter`sees the method as a duplicate code and complains.
The Method cannot be optimised, hence this is required!